controllers/clusteroperator_controller.go

```diff
@@ -67,18 +67,57 @@ type CloudOperatorReconciler struct {
 
 // Reconcile will process the cloud-controller-manager clusterOperator
 func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result, error) {
+	allowedToProvision, err := r.provisioningAllowed(ctx)
+	if err != nil {
+		klog.Errorf("Unable to determine cluster state %v", err)
+		return ctrl.Result{}, err
+	} else if !allowedToProvision {
+		return ctrl.Result{}, nil
+	}
+
+	infra := &configv1.Infrastructure{}
```

Comment:
Since we need the infra in `provisioningAllowed`, why not fetch that at this scope, and pass it into provisioning allowed? I think that would deduplicate the fetching logic here right?

Reply:
+1, that makes sense to me too